### PR TITLE
docs: update PR template

### PR DESCRIPTION
**Pull request by maxsxu**
### Motivation

Providing a more user-friendly contributing experience by making the PR template more structural and clear.

### Modifications

- Added different header sections to describe PR's details

### Verification

Preview this PR itself.

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
I think this was intentionally made concise. cc @alexec

**Comment by maxsxu** [comment]
Open for review and let me know your concerns. Thanks!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+
+### Modifications
+
+<!-- TODO: Say what changes you made. -->
+
+<!-- TODO: Attach screenshots if you changed the UI. -->
+
+### Verification
+
+<!-- TODO: Say how you tested your changes. -->
```
Comment (Joibel) [inline .github/pull_request_template.md:38]: I's suggest that tests should be added or updated here too?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
+- "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
+- Create the PR as draft.
+- Once builds are green, mark your PR "Ready for review".
 
 When changes are requested, please address them and then dismiss the review to get it reviewed again.
 
 -->
+
+
+<!-- Does this PR fixes an issue -->
```
[the remark this answers is not in the document]
Comment (maxsxu) [reply .github/pull_request_template.md]: Thanks, updated

**Comment by terrytangyuan** [comment]
I think it was intentionally concise. See https://github.com/argoproj/argo-workflows/pull/9768 https://github.com/argoproj/argo-workflows/pull/10641

cc @juliev0

File: .github/pull_request_template.md (hunk 1 above)
Comment (juliev0) [reply .github/pull_request_template.md:38]: Although, maybe not a hard requirement if the change is trivial?

**Comment by juliev0** [comment]
> I think it was intentionally concise. See #9768 #10641
> 
> cc @juliev0

I'm fine with the change.

**Review by juliev0 — APPROVED** [review]

**Pull request merged**